test/types/records/copy-elision/copy-elision-sync.chpl

```diff
@@ -0,0 +1,57 @@
+class C { }
+record R {
+  var x: int = 0;
+  var ptr: owned C = new owned C();
+  proc init() {
+    this.x = 0;
+    writeln("init");
+  }
+  proc init(arg:int) {
+    this.x = arg;
+    writeln("init ", arg);
+  }
+  proc init=(other: R) {
+    this.x = other.x;
+    writeln("init= ", other.x);
+  }
+}
+operator R.=(ref lhs:R, rhs:R) {
+  writeln("= ", lhs.x, " ", rhs.x);
+  lhs.x = rhs.x;
+}
+
+
+proc syncSplitInit() {
+  writeln("syncSplitInit");
+  var x: R;
+ 
+  sync {
+    x = new R(1); // split init applies to sync blocks
```

Comment:
It's a little bit confusing for split init tests to occur in `copy-elision-sync.chpl`.

Reply:
Yes, that's true. I did this as an expediency (I basically just pasted the code from the issue into this test), but I think it's OK in this case because split-init and copy-elision are intended to have similar rules; so writing it this way will make it clearer if they start to differ from each other.